doc/app-METHOD.md

```diff
@@ -0,0 +1,80 @@
+<!---
+ Copyright (c) 2016 StrongLoop, IBM, and Express Contributors
+ License: MIT
+-->
+
+<h3 id='app.METHOD'>app.METHOD(path, callback [, callback ...])</h3>
+
+Routes an HTTP request, where METHOD is the HTTP method of the request, such as GET,
+PUT, POST, and so on, in lowercase. Thus, the actual methods are `app.get()`,
+`app.post()`, `app.put()`, and so on.  See below for the complete list.
+
+For more information, see the [routing guide](/guide/routing.html).
```

Comment:
The only way I can think of would be to convert all the links here to links that would work in the repo and then write a post-processing script for the expressjs.com repo that would convert these back to links that would work on the website.  I might be able to figure this out, but it's somewhat beyond my expertise, though I don't expect it would actually be that hard for a real programmer.

But IMO this "scope creep" for this PR, since the point of this is to just move the API docs into this repo, not make them work perfectly in GitHub.  @dougwilson I understand and essentially agree with your suggestion, but could we please do it in a separate PR once this is landed?  

This is getting a bit "meta" but up until now, the .md doc source files have been specifically an _authoring_ format, NOT a _publishing_ format, which is essentially the suggestion here.  This is a significant change, and (as I said) I'm not opposed, but it has a lot of implications.